osmdroid-android/src/main/java/org/osmdroid/tileprovider/modules/DatabaseFileArchive.java

```diff
@@ -70,8 +79,14 @@ public void init(File pFile) throws Exception {
 			final long y = (long) pTile.getY();
 			final long z = (long) pTile.getZoomLevel();
 			final long index = ((z << z) + x << z) + y;
-			final Cursor cur = mDatabase.query(TABLE, tile, COLUMN_KEY+" = " + index + " and "+COLUMN_PROVIDER+" = '" + pTileSource.name() + "'", null, null, null, null);
-
+			
+			Cursor cur;
+			if(!mIgnoreTileSource) {
+				cur = mDatabase.query(TABLE, tile, COLUMN_KEY+" = " + index + " and "+COLUMN_PROVIDER+" = '" + pTileSource.name() + "'", null, null, null, null);
```

Comment:
hmm we should probably change this to use something closer to a prepared statement. this looks like sql injection (not related to this issue obviously)